src/main/java/com/example/cabbooking/servicetest/RideService.java

```diff
@@ -28,42 +36,54 @@ public List<RideDetailsVO> findRide(String username, LocationDetailsVO source, L
         if (user == null) {
             throw new IllegalArgumentException("User not found");
         }
-
-        return driverService.findAvailableDrivers(source,destination,5); // Max distance of 5 units
+        return driverService.findAvailableDrivers(source, destination, 5); // Max distance of 5 units
     }
 
-    public void  chooseRide(String username, String driverName) {
+    public void chooseRide(String username, String driverName) {
         DriverDetailsVO driver = driverService.getDriver(driverName);
+        Ride ride = new Ride();
         if (driver != null && driver.isAvailable()) {
 
             driver.setAvailable(false); // Mark the driver as not available
         } else {
             throw new IllegalArgumentException("Driver not found or not available");
         }
     }
-    public double calculateFare(LocationDetailsVO source,LocationDetailsVO destination){
-        double distance  = calculateDistance(source,destination);
+
+    public double calculateFare(LocationDetailsVO source, LocationDetailsVO destination) {
+        double distance = calculateDistance(source, destination);
         double distanceFare = distance * FareConstant.PER_KM_RATE;
-        double taxAmount =  distanceFare * FareConstant.TAX_RATE;
+        double taxAmount = distanceFare * FareConstant.TAX_RATE;
 
         return FareConstant.BASE_FARE + distanceFare + taxAmount + FareConstant.PLATFORM_CHARGE;
 
     }
 
     private double calculateDistance(LocationDetailsVO source, LocationDetailsVO destination) {
         double xDiff = destination.getXDistance() - source.getXDistance();
-        double yDiff = destination.getYDistance()-source.getYDistance();
+        double yDiff = destination.getYDistance() - source.getYDistance();
         return Math.sqrt(xDiff * xDiff + yDiff * yDiff);
     }
-    public void sortedDriversByDistanceToSource(LocationDetailsVO sources, List<RideDetailsVO>driverDetails){
+
+    public void sortedDriversByDistanceToSource(LocationDetailsVO sources, List<RideDetailsVO> driverDetails) {
         Collections.sort(driverDetails, new Comparator<RideDetailsVO>() {
             @Override
             public int compare(RideDetailsVO o1, RideDetailsVO o2) {
-                double distanceToO1 = calculateDistance(sources,o1.getDriver().getLocation());
-                double distanceToO2 = calculateDistance(sources,o2.getDriver().getLocation());
-                return Double.compare(distanceToO1,distanceToO2);
+                double distanceToO1 = calculateDistance(sources, o1.getDriver().getLocation());
+                double distanceToO2 = calculateDistance(sources, o2.getDriver().getLocation());
+                return Double.compare(distanceToO1, distanceToO2);
             }
         });
     }
 
+    public void cancelRide(String userName, String driverName) {
```

Comment:
is it possible for user to hit this API directly before even requesting for a ride ?this should follow a order right ? first login -> thn select address -> then see fare -> then book -> then cancel how are you maintaining that ?